### PR TITLE
Survey launched

### PR DESCRIPTION
**Pull request by rjweeks**
# Motivation and Context
EQ will send Survey launched event to the events exchange. RM is receive the event and log it for the case.

# What has changed
New event handler for Case Processor

# How to test?
Manually send message via Rabbit
Run the acceptance tests.

# Links
https://trello.com/c/oZbmaF7e/1054-log-surveylaunched-event-3
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=CATD&title=Response+Management+Event+Dictionary
https://github.com/ONSdigital/census-rm-case-api/pull/28

### CODE REVIEW

**Comment by rjweeks** [comment]
[Log SurveyLaunched event (3)](https://trello.com/c/oZbmaF7e/1054-log-surveylaunched-event-3)

File: src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  public SurveyLaunchedReceiver(EventLogger eventLogger, UacService uacService) {
+    this.eventLogger = eventLogger;
+    this.uacService = uacService;
+  }
+
+  @Transactional
+  @ServiceActivator(inputChannel = "surveyLaunchedInputChannel")
+  public void receiveMessage(ResponseManagementEvent event) {
+    if (event.getEvent().getType() == EventTypeDTO.SURVEY_LAUNCHED) {
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java]: write this as `event.getEvent().getType() != EventTypeDTO.SURVEY_LAUNCHED` and then put the throws right after... then you don't need to have the else clause and all the business logic inside the clause, indented

File: src/main/java/uk/gov/ons/census/casesvc/model/dto/ReceiptDTO.java [hunk 2]
```diff
@@ -18,6 +18,8 @@
 
   private boolean unreceipt;
 
+  private String agentId;
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/dto/ReceiptDTO.java]: How does this relate to the story in question?

File: src/test/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiverIT.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            .build();
+
+    // WHEN
+    rabbitQueueHelper.sendMessage(inboundQueue, message);
+
+    // THEN
+    List<Event> events = eventRepository.findAll();
+    assertThat(events.size()).isEqualTo(1);
+    Event event = events.get(0);
+    assertThat(event.getEventDescription()).isEqualTo("SURVEY LAUNCHED");
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiverIT.java]: This ALL CAPS description is not really consistent with other descriptions

File: src/test/java/uk/gov/ons/census/casesvc/testutil/DataUtils.java [hunk 4]
```diff
@@ -91,6 +91,23 @@ public static ResponseManagementEvent getTestResponseManagementReceiptEvent() {
...
+    EventDTO event = managementEvent.getEvent();
+    event.setType(EventTypeDTO.SURVEY_LAUNCHED);
+    event.setSource("CONTACT_CENTRE_API");
+    event.setChannel("CC");
+
+    PayloadDTO payload = managementEvent.getPayload();
+    payload.setUac(null);
+    payload.setCollectionCase(null);
+    payload.setRefusal(null);
+    payload.setPrintCaseSelected(null);
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/census/casesvc/testutil/DataUtils.java]: this payload will still contain a receipt, which we don't really want

File: src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      UacQidLink surveyLaunchedForQid =
+          uacService.findByQid(event.getPayload().getReceipt().getQuestionnaireId());
+      eventLogger.logUacQidEvent(
+          surveyLaunchedForQid,
+          OffsetDateTime.now(),
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java]: This should be the time of the event, from the header

File: src/main/java/uk/gov/ons/census/casesvc/model/dto/ReceiptDTO.java (hunk 2 above)
Comment (rjweeks) [reply src/main/java/uk/gov/ons/census/casesvc/model/dto/ReceiptDTO.java]: Discussed this with @NickGrantONS and renaming this to ResponseDTO.

File: src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java (hunk 1 above)
Comment (rjweeks) [reply src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java]: Done.

File: src/test/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiverIT.java (hunk 3 above)
Comment (rjweeks) [reply src/test/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiverIT.java]: Fixed.

File: src/test/java/uk/gov/ons/census/casesvc/testutil/DataUtils.java (hunk 4 above)
Comment (rjweeks) [reply src/test/java/uk/gov/ons/census/casesvc/testutil/DataUtils.java]: Changed to Response.

File: src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java (hunk 5 above)
Comment (rjweeks) [reply src/main/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiver.java]: Fixed.

File: src/main/java/uk/gov/ons/census/casesvc/config/QueueSetterUpper.java [hunk 6]
```diff
@@ -102,6 +108,11 @@ public Queue rhUacQueue() {
     return new Queue(rhUacQueue, true);
   }
 
+  @Bean
+  Queue surveyLaunchedQueue() {
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/census/casesvc/config/QueueSetterUpper.java]: Should this be `public Queue surveyLaunchedQueue()`?

Comment (ryangrundy7) [reply src/main/java/uk/gov/ons/census/casesvc/config/QueueSetterUpper.java]: Or doesn't it really matter because this file will be gone soon?

Comment (rjweeks) [reply src/main/java/uk/gov/ons/census/casesvc/config/QueueSetterUpper.java]: Yes. Good spot.

**Pull request merged**